scheduler/models.py

```diff
@@ -339,9 +339,10 @@ def _prevent_duplicate_runs(self):
         and decrements that amount from the repeats remaining and updates the scheduled time to the next repeat.
         :return:
         """
-        while self.scheduled_time < timezone.now() and self.repeat and self.repeat > 0:
+        while self.scheduled_time < timezone.now() and (self.repeat is None or (self.repeat and self.repeat > 0)):
```

Comment:
I think it's very important this change to ensure that when repeat is None it will fast-forward the scheduled time into the future, as per the intended use.